landoapi/api/uplift.py

```diff
@@ -23,49 +22,51 @@
 @auth.require_auth0(scopes=("lando", "profile", "email"), userinfo=True)
 def create(data):
     """Create new uplift requests for requested repository & revision"""
+    repo_name, revision_id = data["repository"], data["revision_id"]
 
-    # Validate repository
+    # Validate repository.
     all_repos = get_repos_for_env(current_app.config.get("ENVIRONMENT"))
-    repository = all_repos.get(data["repository"])
+    repository = all_repos.get(repo_name)
     if repository is None or not repository.approval_required:
         return problem(
             400,
-            "No valid uplift repository",
-            "Please select an uplift repository to create that uplift request.",
+            f"Repository {repo_name} is not a repository known to Lando.",
+            "Please select an uplift repository to create the uplift request.",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
+        )
+
+    if not repository.approval_required:
+        return problem(
+            400,
+            f"Repository {repo_name} is not an uplift repository.",
+            "",
             type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
         )
 
     logger.info(
         "Checking approval state",
         extra={
-            "revision": data["revision_id"],
-            "target_repository": data["repository"],
+            "revision": revision_id,
+            "target_repository": repo_name,
         },
     )
-    is_approval, revision, target_repository = check_approval_state(
+    revision, target_repository = check_approval_state(
         g.phabricator,
-        revision_id=data["revision_id"],
-        target_repository_name=data["repository"],
+        revision_id=revision_id,
+        target_repository_name=repo_name,
     )
     logger.info(
         "Approval state is valid",
-        extra={"style": "approval" if is_approval else "uplift"},
+        extra={"style": "uplift"},
     )
```

Comment:
Is there any case when the approval state is not "valid" after checking? Seems like `check_approval_state` always returns something if the revision/repo exist -- so it may be a misnomer (is it just checking whether the repo/revision exist?).

Reply:
I will change the name of the function - in the previous format this function checked for uplift/approval but now it serves a different purpose.